Sources/Plasma/Apps/plPythonPack/PythonInterface.cpp

```diff
@@ -61,61 +65,61 @@ void PythonInterface::initPython(const plFileName& rootDir)
         Py_IgnoreEnvironmentFlag = 1;
         Py_Initialize();
 
-        // intialize any of our special plasma python modules
-//      initP2PInterface();
-        // save object to the Plasma module
-//      plasmaMod = PyImport_ImportModule("Plasma");
-
         // create the StringIO for the stdout and stderr file
         PycStringIO = (struct PycStringIO_CAPI*)PyCObject_Import(const_cast<char*>("cStringIO"), const_cast<char*>("cStringIO_CAPI"));
-        stdFile = (*PycStringIO->NewOutput)(20000);
+        stdOut = (*PycStringIO->NewOutput)(20000);
+        stdErr = (*PycStringIO->NewOutput)(20000);
+
         // if we need the builtins then find the builtin module
         PyObject* sysmod = PyImport_ImportModule("sys");
-        // then add the builtin dicitionary to our module's dictionary
-        if (sysmod != NULL )
+        // then add the builtin dictionary to our module's dictionary
+        if (sysmod != NULL)
         {
             // get the sys's dictionary to find the stdout and stderr
             PyObject* sys_dict = PyModule_GetDict(sysmod);
-            if (stdFile != nil)
+            if (sys_dict != nullptr && stdOut != nullptr && stdErr != nullptr)
             {
-                PyDict_SetItemString(sys_dict, "stdout", stdFile);
-                PyDict_SetItemString(sys_dict, "stderr", stdFile);
+                PyDict_SetItemString(sys_dict, "stdout", stdOut);
+                PyDict_SetItemString(sys_dict, "stderr", stdErr);
```

Comment:
This is really nit-picky, but if `stdOut` is null but `stdErr` is non-null, then `stdErr` is not properly redirected. These should probably be split up behind individual checks since `stdFile` was split.

Reply:
Good idea.